### PR TITLE
Use previous two blocks to calculate difficulty

### PR DESCRIPTION
**Pull request by earlbread**
Currently, we need to traverse from the first block to get the next block difficulty.
This fixes `GetNextBlockDifficulty()` to use only the last two blocks of the chain to calculate the next block difficulty.

### CODE REVIEW

File: Libplanet/Blockchain/Policies/BlockPolicy.cs [hunk 1]
```diff
@@ -131,9 +131,20 @@ DateTimeOffset currentTime
...
-                .First(t => t.Block == null)
-                .Difficulty;
+            int blockCount = blocks.Count();
+
+            if (blockCount <= 1)
+            {
+                return blockCount;
+            }
+
+            Block<T>[] prevBlocks = blocks.Skip(Math.Max(0, blockCount - 2))
```
Comment (longfin) [inline Libplanet/Blockchain/Policies/BlockPolicy.cs]: It seems to be better than before but still a loop with as much complexity as O(n). How about using the index in `BlockChain <T> .Store`?

File: Libplanet/Blockchain/Policies/BlockPolicy.cs [hunk 2]
```diff
@@ -131,9 +131,20 @@ DateTimeOffset currentTime
...
+                .ToArray();
+
+            return GetNextDifficultyFromPrevTimestamp(
+                prevBlocks[0].Timestamp,
+                prevBlocks[1].Timestamp,
+                prevBlocks[1].Difficulty);
         }
 
         private IEnumerable<DifficultyExpectation> ExpectDifficulties(
```
Comment (longfin) [inline Libplanet/Blockchain/Policies/BlockPolicy.cs]: Does this function still used?

Comment (earlbread) [reply Libplanet/Blockchain/Policies/BlockPolicy.cs]: Right, this is still used in `ValidateBlocks()`. I'll find a way to get rid of `ExpectDifficulties()`.

**Comment by earlbread** [comment]
I opened another PR to address other issues.

**Pull request closed without merging**